allennlp/training/trainer.py

```diff
@@ -323,7 +331,9 @@ def _train_epoch(self, epoch: int) -> Dict[str, float]:
 
         # Get tqdm for the training batches
         train_generator = self.iterator(self.train_data, num_epochs=1, shuffle=self.shuffle)
```

Comment:
The `lazy_groups_of` call should be here to create a new generator and `tqdm` should be applied to that. As in https://github.com/allenai/allennlp/blob/3dda5ac9fcf390d8b83eb855249360711707c11c/allennlp/training/trainer.py#L306 previously. Otherwise tqdm will give an inaccurate count of the batches.